envoy-control-runner/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/debug/SnapshotDebugInfo.kt

```diff
@@ -0,0 +1,52 @@
+package pl.allegro.tech.servicemesh.envoycontrol.snapshot.debug
+
+import io.envoyproxy.envoy.api.v2.Cluster
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment
+import io.envoyproxy.envoy.api.v2.Listener
+import io.envoyproxy.envoy.api.v2.RouteConfiguration
+import net.openhft.hashing.LongHashFunction
+
+fun version(version: String) = Version(
+    version,
+    // we use the same method of hashing as Envoy to create comparable string
+    java.lang.Long.toUnsignedString(LongHashFunction.xx().hashBytes(version.toByteArray(Charsets.US_ASCII)))
```

Comment:
@MarcinFalkowski please include links to Envoy source code in places like this. The test based on this started failing after updating to delta XDS and I think the mechanisms changed in some cases.

Reply:
The hash used in Envoy is called xxHash - https://github.com/envoyproxy/envoy/search?p=1&q=xxhash